src/fsharp/ConstraintSolver.fsi

```diff
@@ -94,7 +94,7 @@ type ConstraintSolverState =
 
 type ConstraintSolverEnv 
 
-val BakedInTraitConstraintNames : string list
+val BakedInTraitConstraintNames : System.Collections.Generic.HashSet<string>
```

Comment:
Do we know that this is only used single threaded?  How do we guarantee that moving forward?